syntax = 'proto3';

package main;

message id {
    int64 id = 1;
}

message todo {
    int64 id = 1;
    string value = 2;
}

message todos {
    repeated todo todos = 1;
}

message todoValue{
    string todoValue = 1;
}

message empty {
}

service api {
    rpc addTodo (todoValue) returns (todo);
    rpc getTodo (todoValue) returns (todo);
    rpc getAll (empty) returns (todos);
    rpc delTodo(todoValue) returns (todo);
}


